patroni/dcs/etcd.py

```diff
@@ -264,8 +248,16 @@ def _get_machines_cache_from_srv(self, srv):
 
     def _get_machines_cache_from_dns(self, host, port):
         """One host might be resolved into multiple ip addresses. We will make list out of it"""
-        ret = ['{0}://{1}:{2}'.format(self.protocol, ip, port) for ip in set(self._dns_resolver.resolve(host))]
-        return list(set(ret)) if ret else ['{0}://{1}:{2}'.format(self.protocol, host, port)]
+        if self.protocol == 'http':
+            ret = []
+            for af, _, _, _, sa in self._dns_resolver.resolve(host, port):
+                host, port = sa[:2]
+                if af == socket.AF_INET6:
+                    host = '[{0}]'.format(host)
```

Comment:
I'm not sure IPv6 scopes could be safely ignored. Look at this http://superuser.com/questions/99746/why-is-there-a-percent-sign-in-the-ipv6-address. To me it looks like the local IPv6 address may need a scope to be routed to the correct interface, and omitting it may break  clients connecting to link-local Etcds addresses.